pkg/ring/kvstore/kvstore.go

```diff
@@ -0,0 +1,94 @@
+package kvstore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/snappy"
+)
+
+// KVClient is a high-level client for Consul, that exposes operations
```

Comment:
high-level client for various key value stores, such as Consul or etcd, that exposes operations